### DISCUSSION TITLE
How to intergrate react-hook-form with draft-js [typescript]

### DISCUSSION THREAD

**Discussion by vantu98** [react-hook-form/react-hook-form#6912] [Q&A]
Hi guys.

I'm making a form with 2 fields, **name and description** . name just a text input field, and description is a wysiwyg editor (draft-js). All is written in typescript.

How to implement **editorState** of drat-js with **value** in react-hook-form? Cause 2 type is not match.

This is my codesandbox [demo](https://codesandbox.io/s/integrate-hook-form-editor-wm103?file=/src/App.tsx)

my Editor
```
import React from 'react';

import { Editor, EditorState } from 'draft-js';

import Toolbar from './Toolbar';

interface DraftEditorProps {
  value: EditorState;
  onChange: (editorState: EditorState) => void;
}

const DraftEditor: React.FC<DraftEditorProps> = ({ value, onChange }) => (
  <div className="DraftEditor-container">
    <Toolbar editorState={value} setEditorState={onChange} />

    <Editor editorState={value} textAlignment="left" onChange={onChange} />
  </div>
);

export default DraftEditor;

```

my EditorControl
```
import React from 'react';

import { Controller, useFormContext } from 'react-hook-form';
import { Label } from 'reactstrap';

import DraftEditor from 'components/Editor';

interface EditorControlProps {
  name: string;
  label: string;
}

const EditorControl: React.FC<EditorControlProps> = ({ name, label }) => {
  const { control } = useFormContext();
  return (
    <>
      <Label>{label}</Label>
      <Controller
        name={name}
        control={control}
        render={({ field: { value, onChange } }) => (
          <DraftEditor value={value} onChange={onChange} />
        )}
      />
    </>
  );
};

export default EditorControl;

```

Can you explain and is possible, just send me a codesandbox demo. Thank a lot.

**Comment by bluebill1049** [comment]
can you switch interface to type, and possible provide a codesandbox?

**Reply by vantu98** [reply]
Here you are, this is my [simple demo](https://codesandbox.io/s/integrate-hook-form-editor-wm103?file=/src/App.tsx).
Note: I'm using yup as a resolver for react-hook-form

**Reply by lundn** [reply]
Couldn't get your codesandbox to run before I changed the typing of `description` to `EditorState`, and initialized the default value like:
```
const defaultValues: FormDataState = {
  description: EditorState.createEmpty()
};
```

**Reply by lundn** [reply]
@vantu98 Did that resolve your own issue?

**Reply by vantu98** [reply]
thanks @lundn currently I choose other Editor (TinyMCE) to solve my issues
